Description: call exo-desktop-item-edit instead of gnome-desktop-item-edit when XDG_CURRENT_DESKTOP=XFCE

Index: alacarte-3.5.3/Alacarte/MainWindow.py
===================================================================
--- alacarte-3.5.3.orig/Alacarte/MainWindow.py	2012-06-25 11:59:31.287475573 +1200
+++ alacarte-3.5.3/Alacarte/MainWindow.py	2012-06-25 11:59:39.183475303 +1200
@@ -262,7 +262,10 @@
         else:
             parent = menus[iter][2]
         file_path = os.path.join(util.getUserDirectoryPath(), util.getUniqueFileId('alacarte-made', '.directory'))
-        process = subprocess.Popen(['gnome-desktop-item-edit', file_path], env=os.environ)
+        if os.getenv("XDG_CURRENT_DESKTOP") == "XFCE":
+            process = subprocess.Popen(['exo-desktop-item-edit', file_path], env=os.environ)
+        else:
+            process = subprocess.Popen(['gnome-desktop-item-edit', file_path], env=os.environ)
         GObject.timeout_add(100, self.waitForNewMenuProcess, process, parent.get_menu_id(), file_path)
 
     def on_new_item_button_clicked(self, button):
@@ -275,7 +278,10 @@
         else:
             parent = menus[iter][2]
         file_path = os.path.join(util.getUserItemPath(), util.getUniqueFileId('alacarte-made', '.desktop'))
-        process = subprocess.Popen(['gnome-desktop-item-edit', file_path], env=os.environ)
+        if os.getenv("XDG_CURRENT_DESKTOP") == "XFCE":
+            process = subprocess.Popen(['exo-desktop-item-edit', file_path], env=os.environ)
+        else:
+            process = subprocess.Popen(['gnome-desktop-item-edit', file_path], env=os.environ)
         GObject.timeout_add(100, self.waitForNewItemProcess, process, parent.get_menu_id(), file_path)
 
     def on_new_separator_button_clicked(self, button):
@@ -336,7 +342,10 @@
 
         if file_path not in self.edit_pool:
             self.edit_pool.append(file_path)
-            process = subprocess.Popen(['gnome-desktop-item-edit', file_path], env=os.environ)
+            if os.getenv("XDG_CURRENT_DESKTOP") == "XFCE":
+                process = subprocess.Popen(['exo-desktop-item-edit', file_path], env=os.environ)
+            else:
+                process = subprocess.Popen(['gnome-desktop-item-edit', file_path], env=os.environ)
             GObject.timeout_add(100, self.waitForEditProcess, process, file_path)
 
     def on_menu_tree_cursor_changed(self, treeview):
